gpii/node_modules/flowManager/src/PSPChannel.js

```diff
@@ -97,9 +102,130 @@ fluid.defaults("gpii.pspChannel", {
                 preferencesSavedSuccess: "{gpii.flowManager}.events.preferencesSavedSuccess"
             }
         }
+    },
+    invokers: {
+        generateDefaultSettingControls: {
+            funcName: "gpii.pspChannel.generateDefaultSettingControls",
+            args: ["{flowManager}.defaultSettingsDataPromise", "{that}.options.settingsSchema"]
+        }
     }
 });
 
+/**
+ * Calculate the schemas containing preferences with their default values from the reset to default file.
+ * @param {Object} defaultSettingsDataPromise - The value of flowManager.defaultSettingsDataPromise.
+ * @param {Object} schemas - The content of build/schemas/solution-schema-codex.json.
+ * @return {Object}  - A collection of schemas for each preference defined in the reset to default file.
+ * Each default preference value defined in the reset to default file are populated into the corresponding "default"
+ * field in the output schema. An example:
+ * {
+ *     "http://registry\\.gpii\\.net/common/cursorSize": {
+ *         "schema": {
+ *             "title": "Cursor Size",
+ *             "description": "Cursor size",
+ *             "type": "number",
+ *             "default": 0.8,   // This default value is from the reset to default file
+ *             "minimum": 0,
+ *             "maximum": 1,
+ *             "multipleOf": 0.1
+ *         },
+ *         "liveness": "live"
+ *     },
+ *     ...
+ * }
+ */
+gpii.pspChannel.generateDefaultSettingControls = function (defaultSettingsDataPromise, schemas) {
+    var defaultSettingControls = {};
```

Comment:
On the face of it, this function returns an object which might be very peculiar - a shell whose contents are populated asynchronously. However, "in practice", we know that if the architecture is organised properly, the interlock that prevents a PSP request before defaultSettingsDataPromise is resolved should ensure that this return is actually synchronous. However, it is in everybody's interests for this to be explicit and to cause a hard failure if it is violated. So instead, please do not initialise defaultSettingsControls (note rename) here but leave it blank. Then instead, make a check on line 154 to ensure that it has been initialised before the function returns and issue a fluid.fail() if it has not.